editoast/src/models/auth.rs

```diff
@@ -39,20 +41,44 @@ impl<B: BuiltinRoleSet + Send + Sync> StorageDriver for PgAuthDriver<B> {
     type BuiltinRole = B;
     type Error = AuthDriverError;
 
-    #[tracing::instrument(skip_all, fields(%user), ret, err)]
+    #[tracing::instrument(skip_all, fields(%user_info), ret(level = Level::DEBUG), err)]
+    async fn get_user_id(&self, user_info: &UserInfo) -> Result<Option<i64>, Self::Error> {
+        let conn = self.pool.get().await?;
+        let id = authn_user::table
+            .select(authn_user::id)
+            .filter(authn_user::identity_id.eq(&user_info.identity))
+            .first::<i64>(conn.write().await.deref_mut())
```

Comment:
This is how we access the inner `&mut AsyncPgConnection` that's now wrapped in an `RwLock` in our now opaque `DbConnection`: cf. https://github.com/OpenRailAssociation/osrd/pull/8712.

I suggested implementing `Future` for the connection so that we only have to write `conn.await` instead.

Reply:
My point being, why not `conn.read()`? We might have a design problem that is really misleading in understanding what's going on. If I understand correctly, the `write` basically forward to the `write_owned()` of `RwLock` so I guess there is a possibility to do the same for a `read()`. That said, I didn't test that a read-only would allow us to correctly query the database. If that is not possible, maybe we have a naming problem in the function `write()` 🤷